### PR TITLE
Create Compilers.md

### PR DESCRIPTION
**Pull request by JoeNemo**
Signed-off-by: JoeNemo <joe.devlin@gmail.com>

Signed-off-by: JoeNemo <joe.devlin@gmail.com>

### CODE REVIEW

File: Compilers.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+- Will other non-XPLINK stack support come to xlclang or clang
+
+## Gotcha's
+
+- XLC does allow integer bitfields for all integer widths
+- use "_Packed", not pragma pack(...) - works for all 3 compilers.
+
+## Wishlists
+
+- ZOS Macros for inline ASM in Open XLC (clang)
```
Comment (ifakhrutdinov) [inline Compilers.md:59]: I'd add "support of `thread_local`".

File: Compilers.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+## How do we use xlclang and xlc together?
+
+When projects like zowe/zss are now using some open source (e.g. libyaml), the build compiles the OSS files with xlclang and the zowe-common-c "native" files with xlc and links them together.   Because xlclang only supports AMODE=64 and XPLINK stacks, the xlc builds use the matching options.   They both create standard GOFF object files and the binder (linker for you non-Z folks) has no problems.  
+
+Actually most of zowe-common-c can compile with xlclang (or clang off ZOS).  There are specific issues when files use obsolete non-standard pragmas,etc.  If anybody can find a definitive differences list, please submit a PR to this document.
+
+There are many nits in porting OSS to ZOS.  There are differences, and some omissions in standard libaries.   And a lot of open source code has Linux-isms in it that violate language and POSIX standards.  So it is rarely just "make and run".
+
+## Enter Open XLC 1.1
```
Comment (ifakhrutdinov) [inline Compilers.md]: I know they're pretty sensitive to naming things correctly :-), so perhaps we should use "IBM Open XL C/C++" or "Open XL C/C++"

File: Compilers.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+## Language Standards
+
+As of Zowe 2.0 (2022) feel free to use C11 in most places.  We had been holding at C99 for a long time, but the support for Atomics, type-generics, anonymous struct/union, Unicode, etc. is valuable and becoming more common in OSS.  
+
+# Question, Gotcha's and Wishlists
+
+## Questions
+
+- Will all 3 streams continue indefinitely (XLC,XLCLANG,OPENXLC/CLANG)?
+- Will other non-XPLINK stack support come to xlclang or clang
```
Comment (ifakhrutdinov) [inline Compilers.md:49]: `NR` is deprecated and the doc refers us to the clang documentation. Clang syntax doesn't seem to work. The question is, are there examples of how to use all the `__asm__` features in Open XL C/C++? Are there plans to add examples to the Open XL C/C++ doc?

File: Compilers.md (hunk 1 above)
Comment (JoeNemo) [reply Compilers.md:59]: will add on my next commit.

File: Compilers.md (hunk 2 above)
Comment (JoeNemo) [reply Compilers.md]: Ugh.   It's a sensitive world.

**Review by ifakhrutdinov — APPROVED** [review]
Feel free to accept my suggestions, other than that look good to me.

**Pull request merged**